src/Raven.Client/ServerWide/DatabaseRecord.cs

```diff
@@ -489,6 +496,58 @@ public enum DeletionInProgressStatus
         HardDelete
     }
 
+    public enum MigrationStatus
+    {
+        Pending,
+        Moving,
+        Moved,
+        OwnershipTransferred
+    }
+
+    public class BucketMigration : IDatabaseTask
+    {
+        public MigrationStatus Status;
+        public int Bucket;
+        public int SourceShard;
+        public int DestinationShard;
+        public long MigrationIndex;
+        public long? ConfirmationIndex;
+        public string LastSourceChangeVector;
+
+        public List<string> ConfirmedDestinations = new List<string>();
+        public List<string> ConfirmedSourceCleanup = new List<string>();
```

Comment:
Why are those lists?

Reply:
`ConfirmedDestinations` the destination will confirm all its member nodes in order to move the status of the migration to `OwnershipTransffered` and change the buckets mapping.
`ConfirmedSourceCleanup` the source will confirm the cleanup after the ownership was transferred and only when all source nodes confirmed we will remove this migration from the dictionary

Reply:
For cleanup, I would rather that we do that on the source node, as before, then replicate that normally

Reply:
This part isn't wired yet. Will discuss this for the next PRs